### PR TITLE
fix(serializers): fix up serializer types on user model

### PR DESCRIPTION
**Pull request by JoshFerge**
- fixes some incorrect types
-  adds `kwargs **` as an arg to fix the liskov substitution errors
- types `env.request`

### CODE REVIEW

**Comment by JoshFerge** [comment]
note the openapi-diff test is failing just because the types have been modified slightly

File: src/sentry/api/serializers/models/user.py [hunk 1]
```diff
@@ -144,7 +144,9 @@ def _get_identities(
             results[item.user_id].append(item)
         return results
 
-    def get_attrs(self, item_list: Sequence[User], user: User) -> MutableMapping[User, Any]:
+    def get_attrs(
+        self, item_list: Sequence[User], user: User, **kwargs: Any
```
Comment (asottile-sentry) [inline src/sentry/api/serializers/models/user.py:148]: I'd prefer the type ignore for override than making these signatures overly broad.  or at least raise a `TypeError` if unexpected kwargs are passed

tbh this is "OOP kinda bad" rearing its head

Comment (JoshFerge) [reply src/sentry/api/serializers/models/user.py:148]: gotcha. yeah i was feeling icky about that. May also be able to use some kind of generics on the base class to help with this. i'll iterate on this a bit when i get time.

Comment (JoshFerge) [reply src/sentry/api/serializers/models/user.py:148]: (ah your comment was unrelated to the generic stuff i was thinking about)

**Pull request closed without merging**